zebra-network/src/address_book.rs

```diff
@@ -314,6 +326,34 @@ impl AddressBook {
         meta_addr
     }
 
+    /// Returns true if `updated` has a last_connection_state of `Responded`
+    /// and there is either:
+    /// - no previous addr with the updated.addr IP, or
+    /// - updated has a more recent last_response time than the existing [`MetaAddr`]
+    ///   for that IP in `most_recent_by_ip`.
+    fn should_update_most_recent_by_ip(&self, updated: MetaAddr) -> bool {
+        updated.last_connection_state == PeerAddrState::Responded
+            && updated
+                .last_response()
+                .is_some_and(|updated_last_response| {
+                    self.most_recent_by_ip
+                        .get(&updated.addr.ip())
+                        .map_or(true, |prev| {
+                            prev.last_response() < Some(updated_last_response)
+                        })
+                })
+    }
+
+    /// Returns true if the provided `addr` matches that in `most_recent_by_ip`
+    /// for the IP of the address.
```

Comment:
Nitpick: this doc is a bit unclear to me. I had to read the code to find out it was an exact IP and port match, rather than just an IP address match. ("Address" can mean either.)

It's tricky to explain, here's my attempt at it:
```suggestion
    /// Returns true if `addr` is the latest entry for its IP, which is stored in `most_recent_by_ip`.
    /// The entry is checked for an exact match to the IP and port of `addr`.
```